processor/src/chiplets/memory/mod.rs

```diff
@@ -332,21 +298,31 @@ impl Default for Memory {
 pub struct MemoryLookup {
     ctx: Felt,
     addr: Felt,
-    clk: u32,
+    clk: Felt,
     old_word: Word,
     new_word: Word,
 }
 
 impl MemoryLookup {
-    pub fn new(addr: Felt, clk: u32, old_word: Word, new_word: Word) -> Self {
+    pub fn new(ctx: Felt, addr: Felt, clk: Felt, old_word: Word, new_word: Word) -> Self {
         Self {
-            ctx: ZERO,
+            ctx,
             addr,
             clk,
             old_word,
             new_word,
         }
     }
+
+    pub fn from_ints(ctx: u32, addr: Felt, clk: u32, old_word: Word, new_word: Word) -> Self {
+        Self {
+            ctx: Felt::from(ctx),
+            addr,
+            clk: Felt::from(clk),
+            old_word,
+            new_word,
+        }
+    }
```

Comment:
The issue here is that when requests come from the stack, address is a `Felt` - but we can't use `Felt`'s ask keys in the map since they don't implement the required traits. So, we need to convert `Felt`'s to `u64`. We could change the interface to always take `u64` and then the conversion would happen on the stack side - but I'd like to put more thought in whether this is the best option.

I made a couple of minor changes to make the interfaces a bit more consistent - but this does not fix the underlying issue.

Reply:
Yep, this makes sense. However, I think the point still stands about the `get_value` methods and all the methods that reference it (except `get_old_value`). I think we have a bunch of methods that could possibly be compiled for tests only (`#[cfg(test)]`). There may be a restriction on this due to the integration tests. I'm not sure. But it seems wasteful to be including things if they're only used by tests